apps/courses/models.py

```diff
@@ -0,0 +1,142 @@
+        related_name="main_courses",
+        limit_choices_to={"extended_object__publisher_is_draft": True},
+    )
+    organizations = models.ManyToManyField(
+        "organizations.Organization",
+        related_name="courses",
+        limit_choices_to={"extended_object__publisher_is_draft": True},
+    )
+    subjects = models.ManyToManyField(
+        "Subject",
+        related_name="courses",
+        blank=True,
+        limit_choices_to={"extended_object__publisher_is_draft": True},
+    )
+
+    ROOT_REVERSE_ID = "courses"
+    TEMPLATE_DETAIL = "courses/cms/course_detail.html"
+
+    class Meta:
+        verbose_name = _("course")
+
+    def __str__(self):
+        """Human representation of a course."""
+        session = self.active_session or "no active session"
+        return "{model}: {title} ({session})".format(
+            model=self._meta.verbose_name.title(),
+            title=self.extended_object.get_title(),
+            session=session,
+        )
+
+    def copy_relations(self, oldinstance, language):
+        """
+        We must manually copy the many-to-many relations from the "draft" instance
+        to the "published" instance.
+        """
+        # pylint: disable=no-member
+        self.organizations.set(oldinstance.organizations.all())
+        self.subjects.set(oldinstance.subjects.all())
+
+    def validate_unique(self, exclude=None):
+        """
+        We can't rely on a database constraint for uniqueness because pages
+        exist in two versions: draft and published.
+        """
+        if self.active_session:
+            # Check uniqueness for the version being saved (draft or published)
+            is_draft = self.extended_object.publisher_is_draft
+            uniqueness_query = self.__class__.objects.filter(
+                active_session=self.active_session,
+                extended_object__publisher_is_draft=is_draft,
+            )
+
+            # If the page is being updated, we should exclude it while looking for duplicates
+            if self.pk:
+                uniqueness_query = uniqueness_query.exclude(pk=self.pk)
+
+            # Raise a ValidationError if the active session already exists
+            if uniqueness_query.exists():
+                raise ValidationError(
+                    {
+                        "active_session": [
+                            "A course already exists with this active session."
+                        ]
+                    }
+                )
+        return super().validate_unique(exclude=exclude)
+
+    def save(self, *args, **kwargs):
+        """
+        Enforce validation each time an instance is saved
+        Make sure the main organization is also included in `organizations` as a m2m relation
+        """
+        self.full_clean()
+        super().save(*args, **kwargs)
+
+        if self.pk:
+            # pylint: disable=no-member
+            self.organizations.add(self.main_organization)
```

Comment:
No ordering on the ManyToMany field. 
Adding ordering adds a lot of complexity and my thought was that it is not a good design to use ordering to designate what is the main organization.
So there was 2 reasons to prefer a separate field for that.

I think we also need this field in the index because we want to display the main organization on the list view...